src/apps/btc/btc_sign.c

```diff
@@ -306,7 +307,22 @@ app_btc_sign_error_t app_btc_sign_input(
         return _error(APP_BTC_SIGN_ERR_STATE);
     }
     if (request->sequence != 0xffffffff) {
-        return _error(APP_BTC_SIGN_ERR_INVALID_INPUT);
+        if (request->sequence < 0xffffffff - 2) {
+            return _error(APP_BTC_SIGN_ERR_INVALID_INPUT);
+        }
+        // rbf without a locktime is ill-defined
+        if (request->sequence == 0xffffffff - 2 && _locktime != 0) {
```

Comment:
Consensus enforces the truth values of the following implications:
if locktime > 0 -> nsequence < MAX_INT
if nsequence < MAX_INT - 1 -> RBF
I changed the code to only check for these values. 
This makes me think whether we should let the user verify the nlocktime at all. The rbf check makes sense to an extent, but the nlocktime seems to have little actual impact on the user.

My Opinion: Maybe we should only verify this content in some future "expert" signing mode, since it is data that cannot be verified meaningfully by the hardware wallet after all. However, in light of the things that you showed me today @benma I see no reason to verify it with a normal user. 

I pushed a change to the logic now, that just asserts the above implications, and passes through otherwise.

Reply:
> Consensus enforces the truth values of the following implications:
> if locktime > 0 -> nsequence < MAX_INT

I understood it differently. If nsequence < MAX_INT, locktime is in effect, and it can be any value, including 0.

If nsequence = MAX_INT for all inputs, locktime can be any value, but it is ignored (not compared to current block or time).

I'll have to think about not verifying the locktime with the user at all.